packages/loot-core/src/server/aql/schema/executors.ts

```diff
@@ -32,30 +38,36 @@ function toGroup(parents, children, mapper = x => x) {
 // transactions, and the second will return the count of all parent
 // (or non-split) transactions
 
-function execTransactions(state, query, sql, params, outputTypes) {
-  const tableOptions = query.tableOptions || {};
-  const splitType = tableOptions.splits || 'inline';
+function execTransactions(
+  compilerState: CompilerState,
+  queryState: QueryState,
+  sqlPieces: SqlPieces,
+  params: (string | number)[],
+  outputTypes: OutputTypes,
+) {
+  const tableOptions = queryState.tableOptions || {};
+  const splitType: SplitsOption =
+    (tableOptions.splits as SplitsOption) || 'inline';
 
   if (['all', 'inline', 'none', 'grouped'].indexOf(splitType) === -1) {
     throw new Error(`Invalid “splits” option for transactions: “${splitType}”`);
   }
```

Comment:
Instead of a direct cast could we use a type guard here? i.e.:

```
function isValidSplitsOption(splits: string): splits is SplitsOption {
  return ['all', 'inline', 'none', 'grouped'].includes(splits);
}
```

A bit worried about the potential of a direct cast to drift